packages/react-checkbox/Spec.md

```diff
@@ -2,62 +2,267 @@
+|checked|Checkbox's checked state.|
+|checkmarkIconProps|Icon to be rendered in the checkbox.|
+|className|Additional styles.|
+|componentRef|Optional ref.|
+|defaultChecked|Whether the checkbox should be set to checked by default.|
+|defaultIndeterminate|Whether the checkbox should be set to indeterminate by default.|
+|disabled|Whether the checkbox is disabled or not.|
+|id|Id for the checkbox input.|
+|indeterminate|Checkbox's indeterminate state.|
+|inputProps|Optional props to be applied to the input element before applying other props.|
+|label|String to display next to the checkbox.|
+|name|Name for the checkbox input.|
+|onChange|Event handler to be called after the checked value has changed.|
+|onRenderLabel|Custom render for the label.|
+|required|Required state of the checkbox.|
+|styles|Additional styles.|
+|theme|Additional styles.|
+|title|Title text applied to the root element and the hidden checkbox input.|
+
+### Conclusion
+
+- Most props will follow the approach of `v8` as well as the render structure. The main idea will be to use the native `input` tag and set its opacity to 0 to then render a custom checkbox.
+- This approach will also use the `Label` component from `@fluentui/react-label`
+- The converged `Checkbox` should also support `circular` checkboxes which both `v8` and `v0` do not support.
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```tsx
+<Checkbox label="Example Checkbox"/>
+
+<Checkbox label={<>Example Checkbox with <a href="https://www.microsoft.com">link</a></>} />
+
+<Checkbox circular size="large" label="Circular Checkbox" />
+
+<Checkbox label="Controlled Checkbox" onChange={onChangeFunction} />
+
+<Checkbox indeterminate label="Indeterminate Checkbox" />
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+- A Checkbox has two size variants: `medium (default)` and `large`.
+- A Checkbox has three appearance variants: `disabled`, `unchecked`, `checked`, and `indeterminate`.
+- A Checkbox has a `circular` variant.
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+### Checkbox Props
+
+```tsx
+/**
+ * Checkbox Props
+ */
+export interface CheckboxProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * Label to be rendered with the checkbox.
+   */
+  label?: ShorthandProps<LabelProps>;
+
+  /**
+   * Icon to be displayed when the checkbox is in the checked state.
+   */
+  icon?: React.ReactElement;
+
+  /**
+   * Disabled state of the checkbox.
+   */
+  disabled?: boolean;
+
+  /**
+   * A checkbox can be rendered with a circular shape.
+   */
+  circular?: boolean;
+
+  /**
+   * A checkbox's state can be controlled.
+   * @defaultvalue false
+   */
+  checked?: 'indeterminate' | boolean;
```

Comment:
FYI: before we had an agreement to go with `mixed` as a value to be consistent with `aria-checked` values, https://github.com/microsoft/fluentui/pull/16089#issuecomment-756257494. Are there any arguments to go with

Reply:
I'm no a11y champ, but `input` does support `checked` and `indeterminate` IDL values, I'm not entirely sure why in v0 we went only for `aria-checked` but if we do use native components we should use the `indeterminate` IDL value

Reply:
This PR was mentioned in a meeting with cxe-red yesterday, we decided to follow `aria-checked` and set the `indeterminate` IDL value as well.